annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java

```diff
@@ -0,0 +1,354 @@
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RESOURCE_URI, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RELATION, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK_SEMANTIC, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.LINKING.name().toLowerCase()));
+		assertEquals(storedAnno.getBody().getInternalType(), BodyInternalTypes.GRAPH.name());
+		Graph graphBody = ((GraphBody) storedAnno.getBody()).getGraph();
+		assertNotNull(graphBody.getNode());
+		assertTrue(graphBody.getNode().getHttpUri().equals("http://thesession.org/tunes/52"));
+		assertTrue(graphBody.getRelationName().equals("isSimilarTo"));
+		assertTrue(graphBody.getResourceUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createSemanticTagSpecific() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		
+		String requestBody = getJsonStringInput(SEMANTICTAG_SPECIFIC_STANDARD);
+		
+		Annotation inputAnno = parseTag(requestBody);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isSemanticTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_SPECIFIC_RESOURCE, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.TAGGING.name().toLowerCase()));
+		assertEquals(storedAnno.getBody().getInternalType(), BodyInternalTypes.SEMANTIC_TAG.name());
+		TagBody tagBody = ((SemanticTagBody) storedAnno.getBody());
+		assertNotNull(tagBody.getHttpUri());
+		assertTrue(tagBody.getHttpUri().equals("http://sws.geonames.org/2988506"));
+		assertNotNull(tagBody.getLanguage());
+		assertTrue(tagBody.getLanguage().equals("en"));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createFullTextResourceTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```

Comment:
this is a search not a create test, please rename

Reply:
changed to search...